lib/std/fs.zig

```diff
@@ -116,6 +124,34 @@ pub fn atomicSymLink(allocator: Allocator, existing_path: []const u8, new_path:
     }
 }
 
+/// TODO remove the allocator requirement from this API
+pub fn atomicHardLink(allocator: Allocator, existing_path: []const u8, new_path: []const u8) !void {
```

Comment:
Hmm, this PR is touching too much of Zig's API suface and I am not comfortable in adding any hard link support without prior investigation by me or someone from the core team. Would you mind instead just going back to a simple declarative mechanism like we do with `rosetta`? If not, I'll be happy to take it over and drive it to completion for you. Apologies for chucking out your work on hard links but hopefully my explanation makes sense!

Reply:
Ok, yea that makes sense - no worries. So to be clear, we just want to skip running the macho linker tests (anything that makes a symlink) if `-Denable-windows-symlinks` is not present?